components/net/blob_loader.rs

```diff
@@ -2,83 +2,36 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+use fetch::methods::DoneChannel;
 use filemanager_thread::FileManager;
-use hyper::header::{Charset, ContentLength, ContentType, Headers};
-use hyper::header::{ContentDisposition, DispositionParam, DispositionType};
-use ipc_channel::ipc;
-use mime::{Attr, Mime};
 use net_traits::NetworkError;
 use net_traits::blob_url_store::parse_blob_url;
-use net_traits::filemanager_thread::ReadFileProgress;
+use net_traits::response::{Response, ResponseBody};
 use servo_url::ServoUrl;
+use servo_channel::channel;
 
 // TODO: Check on GET
 // https://w3c.github.io/FileAPI/#requestResponseModel
 
 /// https://fetch.spec.whatwg.org/#concept-basic-fetch (partial)
-// TODO: make async.
-pub fn load_blob_sync
+pub fn load_blob_async
             (url: ServoUrl,
-             filemanager: FileManager)
-             -> Result<(Headers, Vec<u8>), NetworkError> {
+             filemanager: FileManager,
+             done_chan: &mut DoneChannel)
+             -> Response {
     let (id, origin) = match parse_blob_url(&url) {
         Ok((id, origin)) => (id, origin),
         Err(()) => {
-            let e = format!("Invalid blob URL format {:?}", url);
-            return Err(NetworkError::Internal(e));
+            return Response::network_error(NetworkError::Internal("Invalid blob url".into()));
         }
     };
 
-    let (sender, receiver) = ipc::channel().unwrap();
+    let mut response = Response::new(url);
+    let (sender, receiver) = channel();
+    *done_chan = Some((sender.clone(), receiver));
+    *response.body.lock().unwrap() = ResponseBody::Receiving(vec![]);
     let check_url_validity = true;
-    filemanager.read_file(sender, id, check_url_validity, origin);
+    filemanager.fetch_file(sender, id, check_url_validity, origin, &mut response);
```

Comment:
Since the only caller of this method is passing true for check_url_validity, let's remove the argument.